mobile/apps/photos/lib/utils/file_uploader.dart

```diff
@@ -434,6 +434,15 @@ class FileUploader {
     if (Platform.isAndroid) {
       final bool hasPermission = await Permission.accessMediaLocation.isGranted;
       if (!hasPermission) {
+        // In background mode, we can't request permissions (no UI available)
+        // Skip the permission request and continue with upload
+        if (_processType == ProcessType.background) {
+          _logger.info(
+            "Media location access not granted, but continuing in background mode",
+          );
+          return;
```

Comment:
We shouldn't continue upload if media location access is not granted. Instead of returning, throw `NoMediaLocationAccessError()`